src/modules/job-archive/job-archive.c

```diff
@@ -543,10 +555,11 @@ int mod_main (flux_t *h, int ac, char **av)
     if (!ctx)
         return -1;
 
-    process_config (ctx);
+    if (process_config (ctx) < 0)
+        goto done;
 
-    /* we do nothing if no dbpath specified */
-    if (ctx->dbpath) {
+    /* we do nothing if period not set */
+    if (ctx->period > 0.0) {
         if (job_archive_init (ctx) < 0)
             goto done;
```

Comment:
Shouldn't it be an error if the period is not set?  Is loading the module but having it do nothing a valid use case?

Reply:
ahh yes, I was keeping the style that was previously there, but we should error out instead.